variable/reduction.cpp

```diff
@@ -3,39 +3,51 @@
 /// @file
 /// @author Simon Heybrock
 #include "scipp/variable/reduction.h"
+#include "scipp/common/reduction.h"
 #include "scipp/core/dtype.h"
 #include "scipp/core/element/arithmetic.h"
 #include "scipp/core/element/comparison.h"
 #include "scipp/core/element/logical.h"
 #include "scipp/variable/arithmetic.h"
-#include "scipp/variable/except.h"
 #include "scipp/variable/transform.h"
 
-#include "operations_common.h"
-
 using namespace scipp::core;
+using scipp::common::reduce_all_dims;
 
 namespace scipp::variable {
 
+// Workaround VS C7526 (undefined inline variable) with dtype<> in template.
+bool is_dtype_bool(const VariableConstView &var) {
+  return var.dtype() == dtype<bool>;
+}
+bool is_dtype_int64(const VariableConstView &var) {
```

Comment:
Thanks for the fix @Matthew-Andrew. Is this int function necessary? I didn't see it complain about dtype<int> in the failed build.

Reply:
Locally, once I fixed the bool complaint I started seeing an identical error for the use of `d_type<Int64_t>`